pkg/infrastructure/azure/compute.go

```diff
@@ -111,6 +112,17 @@ type CreateGalleryImageOutput struct {
 func CreateGalleryImage(ctx context.Context, in *CreateGalleryImageInput) (*CreateGalleryImageOutput, error) {
 	logrus.Debugf("Creating gallery image: %s", in.GalleryImageName)
 
+	// Set `Features` within the image properties of Gen V2 images
+	// so that they can be used when encryptionAtHost is enabled.
+	var features []*armcompute.GalleryImageFeature
+	if in.SecurityType != "" {
+		feature := armcompute.GalleryImageFeature{
+			Name:  to.Ptr("SecurityType"),
+			Value: to.Ptr(in.SecurityType),
+		}
+		features = append(features, to.Ptr(feature))
+	}
+
```

Comment:
Currently, the fix takes a safe approach by populating the `features` only when `SecurityType` is enabled in the install-config. We could also set it always whether that feature is enabled or not.